ext/src/Duration.c

```diff
@@ -9,34 +9,31 @@
 
 zend_class_entry *php_driver_duration_ce = NULL;
 
-static void to_string(zval *result, cass_int64_t value)
+static void to_string(zval *result, cass_int32_t value)
 {
-  // Adapted from Bigint to_string function.
   char *string;
-#ifdef WIN32
-  spprintf(&string, 0, "%I64d", (long long int) value);
-#else
-  spprintf(&string, 0, "%lld", (long long int) value);
-#endif
+  spprintf(&string, 0, "%d", value);
   PHP5TO7_ZVAL_STRING(result, string);
   efree(string);
 }
 
-static int get_int64(zval* value, cass_int64_t* destination, const char* param_name TSRMLS_DC)
+static int get_int32(zval* value, cass_int32_t* destination, const char* param_name TSRMLS_DC)
 {
   // Adapted from Bigint __construct method.
   if (Z_TYPE_P(value) == IS_LONG) {
-    *destination = (cass_int64_t) Z_LVAL_P(value);
+    *destination = Z_LVAL_P(value);
   } else if (Z_TYPE_P(value) == IS_DOUBLE) {
-    *destination = (cass_int64_t) Z_DVAL_P(value);
+    *destination = (cass_int32_t) Z_DVAL_P(value);
   } else if (Z_TYPE_P(value) == IS_STRING) {
-    if (!php_driver_parse_bigint(Z_STRVAL_P(value), Z_STRLEN_P(value), destination TSRMLS_CC)) {
+    cass_int64_t parsed_big_int;
+    if (!php_driver_parse_bigint(Z_STRVAL_P(value), Z_STRLEN_P(value), &parsed_big_int TSRMLS_CC)) {
       return 0;
     }
+    *destination = (cass_int32_t) parsed_big_int;
```

Comment:
I thought about that a bit, but at the time I figured that Bigint doesn't do any special checks when taking longs/doubles. But now that I think about it, Bigint doesn't suffer with this problem since it's storing its value as a 64-bit number.

So I agree we should catch this overflow and error out, and similarly error out in the clause above where we deal with doubles.